File: python/triangle/triangle.py
def equilateral(sides):
    a, b, c = sides
    return (a == b == c) and is_triangle(a, b, c)


def isosceles(sides):
    a, b, c = sides
    return (a == b or a == c or b == c) and is_triangle(a, b, c)


def scalene(sides):
    a, b, c = sides
    return a != b and a != c and b != c and is_triangle(a, b, c)


def is_triangle(a, b, c):
    return a + b >= c and b + c >= a and a + c >= b and a > 0 and b > 0 and c > 0

File: prolog/triangle/triangle.pl
triangle(A, B, C, T) :-
	A + B > C,
	B + C > A,
	C + A > B,
	triangle_(A, B, C, T).

triangle_(A, A, A, "equilateral") :- !.
triangle_(A, B, C, "isosceles") :- (A =:= B; B =:= C; C =:= A), !.
triangle_(A, B, C, "scalene") :- A =\= B, B =\= C, C =\= A, !.
